index.js

```diff
@@ -3983,6 +3983,26 @@
   S.splitOn =
   def('splitOn', {}, [$.String, $.String, $.Array($.String)], splitOn);
 
+  //# splitOnRegex :: RegExp -> String -> Array String
+  //.
+  //. Returns the substrings of its second argument separated by evaluated
+  //. matches of its first argument.
+  //.
+  //. See also [`splitOn`](#splitOn).
+  //.
+  //. ```javascript
+  //. > S.splitOnRegex(/\s+/, 'foo  bar baz   qux')
+  //. ['foo', 'bar', 'baz', 'qux']
+  //. ```
+  function splitOnRegex(pattern, s) {
+    return s.split(pattern);
+  }
+  S.splitOnRegex =
+  def('splitOnRegex',
+    {},
+    [$.RegExp, $.String, $.Array($.String)],
+    splitOnRegex);
```

Comment:
To be clear, I'm suggesting reusing the internal `splitOn` function as the implementation for `S.splitOnRegex`, without touching `S.splitOn`. We'd have something like this:

```javascript
function splitOn(separator, s) {
  return s.split(separator);
}
S.splitOn =
def('splitOn', {}, [$.String, $.String, $.Array($.String)], splitOn);

S.splitOnRegex =
def('splitOnRegex', {}, [$.RegExp, $.String, $.Array($.String)], splitOn);
```